src/Http/Controllers/Auth/SocialiteController.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core\Http\Controllers\Auth;
+
+use Blumilk\Meetup\Core\Http\Controllers\Controller;
+use Blumilk\Meetup\Core\Models\User;
+use Illuminate\Http\RedirectResponse;
+use Laravel\Socialite\Facades\Socialite;
+
+class SocialiteController extends Controller
+{
+    public function redirectToGoogle(): RedirectResponse
+    {
+        return Socialite::driver("google")->redirect();
+    }
+    public function handleGoogleCallback(): RedirectResponse
+    {
+        $user = Socialite::driver("google")->user();
+
+        $this->registerOrLogin($user);
+
+        return redirect()->route("home");
+    }
+    public function redirectToFacebook(): RedirectResponse
+    {
+        return Socialite::driver("facebook")->redirect();
+    }
+    public function handleFacebookCallback(): RedirectResponse
+    {
+        $user = Socialite::driver("google")->user();
+
+        $this->registerOrLogin($user);
+
+        return redirect()->route("home");
+    }
+
+    protected function registerOrLogin($data)
+    {
+        $user = User::where("email", $data["email"])->first();
+        if (!$user) {
+            $user = User::create([
+                "name" => $data["name"],
+                "email" => $data["email"],
+                "provider_id" => $data["id"],
+            ]);
+        }
```

Comment:
Instead of this, you can use `firstOrCreate` for eloquent models - https://laravel.com/docs/9.x/eloquent#retrieving-or-creating-models.